### PR TITLE
Update tracking CI, add new CI action for LTO and modify other actions to use just

### LINKED ISSUE #1459: Updates to CI: use just, update tracking, enforce clang-tidy
**Is your feature request related to a problem? Please describe.**
 - the CI compilion still uses the old approach
 - we updated the tracking configs
 - after https://github.com/LDMX-Software/ldmx-sw/issues/1176 is resolved, we should enforce those settings for the future code

**Describe the solution you'd like**
 - move CI compile to `just`
 - propagate new tracking config to the CI config too
 - enforce clang-tidy at CI level

**Describe alternatives you've considered**

We could discuss the last point at a software meeting, alternative seems more complicated when we do compile log diffs, in the end the developer should fix those problems

### PR DESCRIPTION
**Pull request by tvami**
I am updating _ldmx-sw_, here are the details.

### What are the issues that this addresses?
Resolves https://github.com/LDMX-Software/ldmx-sw/issues/1459

## Check List
- [x] I successfully compiled _ldmx-sw_ with my developments
- [x] I ran my developments and the following shows that they are successful.

Well the test of the pudding is to see what github does with these actions:
 * it does work according to the expectation

### CODE REVIEW

**Review by tomeichlersmith — COMMENTED** [review]
This is great :) definitely a step in the right direction.

I don't really have any notes except a question. Do you think it would be helpful to summarize the default ci building in a just recipe to avoid copying it around?

```
[private]
ci-build: install-denv init (configure-force-error) build
```
(The parentheses help inform `just` that we aren't giving any more arguments to the configure step which would normally swallow all following arguments as build configuration.)

Then the CI calls would be
```yaml
run: just ci-build
```

might be helpful, might not be.

**Comment by tvami** [comment]
Hmm, I dont know, I see your point that this would certainly be much more compact, but I kinda like that it's spelled out, just for making it more obvious. And also it's more segmented right now in case some of the prev steps failed. So I'm kinda leaning forward keeping it as is.

**Comment by tvami** [comment]
Compilation part looks good! Running the pn validation here
https://github.com/LDMX-Software/ldmx-sw/actions/runs/10911000232

**Comment by tvami** [comment]
> Compilation part looks good! Running the pn validation here https://github.com/LDMX-Software/ldmx-sw/actions/runs/10911000232

Yaay, ecalPN is back to work!

**Review by danyi211 — APPROVED** [review]

File: .github/workflows/pr_validation.yml [hunk 1]
```diff
@@ -23,10 +23,18 @@ jobs:
...
+ 
     - name: Compile and Install ldmx-sw
-      uses: ./.github/actions/setup
+      run: |
+        just install-denv init configure-force-error
+        just build
+      shell: bash
 
-    - name: Git Gold Histograms from trunk
+    - name: Get Gold Histograms from trunk
```
Comment (tomeichlersmith) [inline .github/workflows/pr_validation.yml]: This mis-spelling was on purpose ;) but I recognize that might be confusing for some

**Review by tomeichlersmith — APPROVED** [review]

File: .github/workflows/pr_validation.yml (hunk 1 above)
Comment (tvami) [reply .github/workflows/pr_validation.yml]: I can put it back, I just did that change so there is a commit to trigger tests

**Pull request merged**